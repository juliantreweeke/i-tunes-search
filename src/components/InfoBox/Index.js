import React from "react";
import PropTypes from "prop-types";
import Button from "../Button/Index";
import Heading from "../Heading/Index";
import styles from "./infoBox.module.css";

const InfoBox = ({ buttonText, data, headingLevel }) => (
  <div className={styles.infoBox}>
    <Heading
      size="massive"
      className={styles.heading}
      headingLevel={headingLevel}
    >
      {data.title}
    </Heading>
    {data.text && (
      <p className={styles.text}>
        <strong>{data.text}</strong> - {data.detail}
      </p>
    )}
    {data.url && (
      <div className={styles.buttonContainer}>
        <Button href={data.url} label={data.title}>
          {buttonText}
        </Button>
      </div>
    )}
  </div>
);

InfoBox.propTypes = {
  buttonText: PropTypes.string,
  data: PropTypes.object,
  headingLevel: PropTypes.string,
};

export default InfoBox;
