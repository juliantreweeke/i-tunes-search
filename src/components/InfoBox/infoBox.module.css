.infoBox {
  text-align: center;
  margin-bottom: var(--space-md);
}

.infoBox .text {
  font-size: var(--text-md);
}

.infoBox .buttonContainer {
  margin-top: auto;
}
