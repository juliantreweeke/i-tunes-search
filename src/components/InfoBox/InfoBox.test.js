import React from "react";
import { render } from "@testing-library/react";
import InfoBox from "./Index";
import { axe, toHaveNoViolations } from "jest-axe";

const mockData = {
  heading: "heading",
  title: 'a title',
  image: "testimage.com",
  text: "some text for testing",
  detail: "some detail text for testing",
};
const mockUrl = "test.com";
const mockButtonText = "buy now";

describe("InfoBox", () => {
  it("should render", () => {
    const { container } = render(
      <InfoBox
        data={mockData}
        url={mockUrl}
        headingLevel="h2"
        buttonText={mockButtonText}
      />
    );
    expect(container).toBeTruthy();
  });

  it("matches snapshot", () => {
    const { container } = render(
      <InfoBox
        data={mockData}
        url={mockUrl}
        headingLevel="h2"
        buttonText={mockButtonText}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("Component should have no accessibility violations", async () => {
    const { container } = render(
      <InfoBox
        data={mockData}
        url={mockUrl}
        headingLevel="h2"
        buttonText={mockButtonText}
      />
    );
    expect.extend(toHaveNoViolations);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
