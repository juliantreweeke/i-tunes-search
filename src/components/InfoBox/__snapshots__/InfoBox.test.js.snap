// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBox matches snapshot 1`] = `
<div>
  <div
    class="infoBox"
  >
    <h2
      class="heading"
      data-size="massive"
    >
      a title
    </h2>
    <p
      class="text"
    >
      <strong>
        some text for testing
      </strong>
       - 
      some detail text for testing
    </p>
  </div>
</div>
`;
